//
// PROJECT:         Aspia
// FILE:            proto/system_info/printers.proto
// LICENSE:         Mozilla Public License Version 2.0
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message Programs
{
    message Item
    {
        string name             = 1;
        string version          = 2;
        string publisher        = 3;
        string install_date     = 4;
        string install_location = 5;
    }

    repeated Item item = 1;
}

message Services
{
    message Item
    {
        string name         = 1;
        string display_name = 2;
        string description  = 3;

        enum Status
        {
            STATUS_UNKNOWN          = 0;
            STATUS_CONTINUE_PENDING = 1;
            STATUS_PAUSE_PENDING    = 2;
            STATUS_PAUSED           = 3;
            STATUS_RUNNING          = 4;
            STATUS_START_PENDING    = 5;
            STATUS_STOP_PENDING     = 6;
            STATUS_STOPPED          = 7;
        }

        Status status = 4;

        enum StartupType
        {
            STARTUP_TYPE_UNKNOWN      = 0;
            STARTUP_TYPE_AUTO_START   = 1;
            STARTUP_TYPE_DEMAND_START = 2;
            STARTUP_TYPE_DISABLED     = 3;
            STARTUP_TYPE_BOOT_START   = 4;
            STARTUP_TYPE_SYSTEM_START = 5;
        }

        StartupType startup_type = 5;
        string binary_path       = 6;
        string start_name        = 7;
    }

    repeated Item item = 1;
}

message OpenConnections
{
    enum Protocol
    {
        PROTOCOL_UNKNOWN = 0;
        PROTOCOL_TCP     = 1;
        PROTOCOL_UDP     = 2;
    }

    enum State
    {
        STATE_UNKNOWN     = 0;
        STATE_CLOSED      = 1;
        STATE_LISTENING   = 2;
        STATE_SYN_SENT    = 3;
        STATE_SYN_RCVD    = 4;
        STATE_ESTABLISHED = 5;
        STATE_FIN_WAIT1   = 6;
        STATE_FIN_WAIT2   = 7;
        STATE_CLOSE_WAIT  = 8;
        STATE_CLOSING     = 9;
        STATE_LAST_ACK    = 10;
        STATE_TIME_WAIT   = 11;
        STATE_DELETE_TCB  = 12;
    }

    message Item
    {
        string process_name   = 1;
        Protocol protocol     = 2;
        string local_address  = 3;
        string remote_address = 4;
        uint32 local_port     = 5;
        uint32 remote_port    = 6;
        State state           = 7;
    }

    repeated Item item = 1;
}

message SharedResources
{
    message Item
    {
        string name = 1;

        enum Type
        {
            TYPE_UNKNOWN   = 0;
            TYPE_DISK      = 1;
            TYPE_PRINTER   = 2;
            TYPE_DEVICE    = 3;
            TYPE_IPC       = 4;
            TYPE_SPECIAL   = 5;
            TYPE_TEMPORARY = 6;
        }

        Type type           = 2;
        string description  = 3;
        string local_path   = 4;
        uint32 current_uses = 5;
        uint32 maximum_uses = 6;
    }

    repeated Item item = 1;
}

message Routes
{
    message Item
    {
        string destonation = 1;
        string mask        = 2;
        string gateway     = 3;
        uint32 metric      = 4;
    }

    repeated Item item = 1;
}

message Users
{
    message Item
    {
        string name                  = 1;
        string full_name             = 2;
        string comment               = 3;
        bool is_disabled             = 4;
        bool is_password_cant_change = 5;
        bool is_password_expired     = 6;
        bool is_dont_expire_password = 7;
        bool is_lockout              = 8;
        uint32 number_logons         = 9;
        uint32 bad_password_count    = 10;
        int64 last_logon_time        = 11;
    }

    repeated Item item = 1;
}

message UserGroups
{
    message Item
    {
        string name    = 1;
        string comment = 2;
    }

    repeated Item item = 1;
}

message Sessions
{
    enum ConnectState
    {
        CONNECT_STATE_UNKNOWN       = 0;
        CONNECT_STATE_ACTIVE        = 1;
        CONNECT_STATE_CONNECTED     = 2;
        CONNECT_STATE_CONNECT_QUERY = 3;
        CONNECT_STATE_SHADOW        = 4;
        CONNECT_STATE_DISCONNECTED  = 5;
        CONNECT_STATE_IDLE          = 6;
        CONNECT_STATE_LISTEN        = 7;
        CONNECT_STATE_RESET         = 8;
        CONNECT_STATE_DOWN          = 9;
        CONNECT_STATE_INIT          = 10;
    }

    message Item
    {
        string user_name           = 1;
        string domain_name         = 2;
        uint32 session_id          = 3;
        ConnectState connect_state = 4;
        string client_name         = 5;
        string winstation_name     = 6;
    }

    repeated Item item = 1;
}
